### ISSUE TITLE
Escaping template strings not working

### ISSUE THREAD

**Issue by hnicke** [garden-io/garden#3989]
### Current Behavior

I need the literal value `${foo}` in a resulting k8s template of a garden kubernetes module.
The [garden docs](https://docs.garden.io/using-garden/module-templates#escaping-template-strings) describe how this should be done: Using `$${}` instead of `${}`.

```yaml
...
command: [ bash ]
args:
  - c
  - |
    ....
    -d '$${foo}'
```

However, it ends up literally like this after rendering:
```yaml
...
command: [ bash ]
args:
  - c
  - |
    ....
    -d '$${foo}'
```

I.e., no interpolation at all happened.

Garden doesn't eat the dollar. Maybe this only happens in the free tier :laughing: 

### Expected behavior
I expect this to be the result of the template rendering:
```yaml
...
command: [ bash ]
args:
  - c
  - |
    ....
    -d '${foo}'
```

### Additional idea
@worldofgeese pointed me to the docs, which I haven't found on my own.
I suggest moving the whole `Escaping template strings` paragraph from `Module Templates` to `Variables and Templating`. My understanding is that the `Module Templates` page only holds information in regards to the `kind: Module; type: templated` modules. The interpolation escape mechanism however is a globally available feature, right?

### Your environment

* **OS:** arch linux
* **How I'm running Kubernetes:** GKE
* 
`garden version`
v0.12.53

**Comment by orymate** [comment]
We've workarounded this by rearranging the manifest templates in a way that all variable substitutions come before the parts that need escaping, and adding a comment like `# hack: $${}` between the two. No variable substitutions are done in the second part of the manifest, at least in 0.12.60.

```yaml
...
spec:
  loggingRef: ${environment.namespace}-demo  # TODO garden fails to expand this variable after the first escape, keeping this line before that
  # hack: $${}
  filters:
    - parser:
        metrics-probe:
          key: "probe"
          labels:
            namespace: "${json#kubernetes#namespace_name}"  # the manifests contain these exact label templates that use similar variable expansion syntax as garden
            pod: "${json#kubernetes#pod_name}"
```

**Comment by vvagaytsev** [comment]
@orymate thanks for sharing the workaround!

**Comment by vvagaytsev** [comment]
@orymate we have fixed this in [0.13.39](https://github.com/garden-io/garden/releases/tag/0.13.39). Feel free to try it out!

Please note, that the [workaround](https://github.com/garden-io/garden/issues/3989#issuecomment-1672955747) cannot be used along with 0.13.39.
